packages/next/client/index.tsx

```diff
@@ -605,20 +604,30 @@ function AppContainer({
   )
 }
 
-const wrapApp = (App: AppComponent) => (
-  wrappedAppProps: Record<string, any>
-): JSX.Element => {
-  const appProps: AppProps = {
-    ...wrappedAppProps,
-    Component: CachedComponent,
-    err: hydrateErr,
-    router,
+function wrapApp(
+  App: AppComponent,
+  Component: React.ComponentType,
+  ctx: NextPageContext
+) {
+  const mockRouter = {
```

Comment:
I don't think we want a *fake* router, but a real router, just with different data. That's essentially what `makePublicRouterInstance` does.